### PR TITLE
Add a configurable scale factor to scale in/out

### PR DESCRIPTION
**Pull request by lox**
This adds a scale-in and scale-out factor that can be used to speed up or slow down scale outs and ins. 

For instance, setting `SCALE_IN_FACTOR=0.5` will reduce the change applied in each scale in by 50%.  Setting `SCALE_OUT_FACTOR=2.0` will increase each scale out change by 200%.

This is a simpler take on #12.

### CODE REVIEW

**Comment by lox** [comment]
@etaoins would love to hear your thoughts!

File: lambda/main.go [hunk 1]
```diff
@@ -28,54 +29,64 @@ func main() {
...
 
-	if scaleInAdjustmentStr := os.Getenv(`SCALE_IN_ADJUSTMENT`); scaleInAdjustmentStr != "" {
-		var err error
-		scaleInAdjustment, err = strconv.ParseInt(scaleInAdjustmentStr, 10, 64)
-		if err != nil {
+	if v := os.Getenv(`SCALE_IN_FACTOR`); v != "" {
+		if scaleInFactor, err = strconv.ParseFloat(v, 64); err != nil {
 			return "", err
 		}
+		scaleInFactor = math.Abs(scaleInFactor)
```
Comment (etaoins) [inline lambda/main.go]: Nice usability tweak :+1:

File: scaler/scaler.go [hunk 2]
```diff
@@ -106,51 +109,136 @@ func (s *Scaler) Run() error {
...
 
-		minimumDesired := current.DesiredCount + s.scaleInParams.Adjustment
-		if desired < minimumDesired {
-			desired = minimumDesired
+	// Calculate the change in the desired count, will be negative
+	change := desired - current.DesiredCount
+
+	// Apply scaling factor if one is given
+	if factor := s.scaleInParams.Factor; factor != 0 {
+		// Use Floor to avoid never reaching upper bound
```
Comment (etaoins) [inline scaler/scaler.go]:
```suggestion
		// Use Floor to avoid never reaching lower bound
```

File: scaler/scaler.go [hunk 3]
```diff
@@ -106,51 +109,136 @@ func (s *Scaler) Run() error {
...
+		factoredChange := int64(math.Floor(float64(change) * factor))
+
+		if factoredChange < change {
+			log.Printf("👮‍️ Increasing scale-in of %d by factor of %0.2f",
```
Comment (etaoins) [inline scaler/scaler.go]: I'm not sure about the utility of having a scale-in factor >100% but if someone is crazy enough to configure that I guess we should honour it.

Comment (lox) [reply scaler/scaler.go]: Yeah, I thought the same 🤷🏼‍♂️

**Pull request merged**